File: lib/src/models/pose.dart
class Muscle {
  int id;
  String name;
  String image;
  String description;
  String created_at;
  String updated_at;
  int active_status;

  Muscle({
    required this.id,
    required this.name,
    required this.image,
    required this.description,
    required this.created_at,
    required this.updated_at,
    required this.active_status,
  });
}

class Pose {
  int id;
  List<Muscle> muscles;
  String name;
  String image_url;
  int duration;
  int calories;
  String keypoint_url;
  String instruction;
  String created_at;
  String updated_at;
  int active_status;
  int level;

  Pose({
    required this.id,
    required this.muscles,
    required this.name,
    required this.image_url,
    required this.duration,
    required this.calories,
    required this.keypoint_url,
    required this.instruction,
    required this.created_at,
    required this.updated_at,
    required this.active_status,
    required this.level,
  });
}

File: android/app/src/main/kotlin/com/example/yogi_application/model/Pose.kt
package com.yogitech.yogi_application.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.json.Json
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString


@Serializable
data class Pose(
    val id: Int,
    val muscles: List<Muscle>,
    val name: String,
    @SerialName("image_url" ) val imageUrl: String,
    val duration: Int,
    val calories: String,
    @SerialName("keypoint_url") val keypointUrl: String,
    val instruction: String,
            @SerialName("created_at") val createdAt: String,
                @SerialName("updated_at") val updatedAt: String,
                    @SerialName("active_status") val activeStatus: Int,
    val level: Int
) {
    companion object {
        fun fromJson(source: String): Pose {
            return Json.decodeFromString(source)
        }
    }

    fun toJson(): String {
        return Json.encodeToString(this)
    }
}
